File: TeRKPeerCommon/code/java/communicator/src/edu/cmu/ri/mrpl/TeRK/servants/Servants.java
package edu.cmu.ri.mrpl.TeRK.servants;

import java.util.Collections;
import java.util.HashSet;
import java.util.Set;
import Ice.ObjectPrx;
import edu.cmu.ri.mrpl.TeRK.TerkUserPrx;
import edu.cmu.ri.mrpl.peer.ConnectionEventHandlerPrx;

/**
 * @author Chris Bartley (bartley@cmu.edu)
 */
public final class Servants
   {
   private final TerkUserPrx mainServantProxy;
   private final ConnectionEventHandlerPrx connectionEventHandlerProxy;
   private final Set<ObjectPrx> secondaryServantProxies = new HashSet<ObjectPrx>();

   public Servants(final TerkUserPrx mainServantProxy,
                   final ConnectionEventHandlerPrx connectionEventHandlerProxy)
      {
      this(mainServantProxy, connectionEventHandlerProxy, null);
      }

   public Servants(final TerkUserPrx mainServantProxy,
                   final ConnectionEventHandlerPrx connectionEventHandlerProxy,
                   final Set<ObjectPrx> secondaryServantProxies)
      {
      this.mainServantProxy = mainServantProxy;
      this.connectionEventHandlerProxy = connectionEventHandlerProxy;
      if ((secondaryServantProxies != null) && (!secondaryServantProxies.isEmpty()))
         {
         this.secondaryServantProxies.addAll(secondaryServantProxies);
         }
      }

   public TerkUserPrx getMainServantProxy()
      {
      return mainServantProxy;
      }

   public ConnectionEventHandlerPrx getConnectionEventHandlerProxy()
      {
      return connectionEventHandlerProxy;
      }

   public Set<ObjectPrx> getSecondaryServantProxies()
      {
      return Collections.unmodifiableSet(secondaryServantProxies);
      }
   }

File: TeRKPeerCommon/code/c#/src/TeRK/communications/Servants.cs
using System;
using System.Collections.Generic;
using System.Collections;
using System.Text;
using System.Text.RegularExpressions;
using System.Reflection;
using System.Threading;
using Ice;
using peer;
using TeRK.components.servants;
using TeRK.components.services;
using Exception = System.Exception;
using RouterPrx = Glacier2.RouterPrx;
using RouterPrxHelper = Glacier2.RouterPrxHelper;

namespace TeRK.communications
{
    public class Servants
    {
        private TerkUserPrx mainServantProxy;
        private ConnectionEventHandlerPrx connectionEventHandlerProxy;
        private List<ObjectPrx> secondaryServantProxies = new List<ObjectPrx>();

        public Servants(TerkUserPrx mainServantProxy,
                        ConnectionEventHandlerPrx connectionEventHandlerProxy) :
            this(mainServantProxy, connectionEventHandlerProxy, null)
        {
        }

        public Servants(TerkUserPrx mainServantProxy,
                        ConnectionEventHandlerPrx connectionEventHandlerProxy,
                        List<ObjectPrx> secondaryServantProxies)
        {
            this.mainServantProxy = mainServantProxy;
            this.connectionEventHandlerProxy = connectionEventHandlerProxy;
            if ((secondaryServantProxies != null) && (secondaryServantProxies.Count > 0))
            {
                this.secondaryServantProxies.AddRange(secondaryServantProxies);
            }
        }

        public TerkUserPrx getMainServantProxy()
        {
            return mainServantProxy;
        }

        public ConnectionEventHandlerPrx getConnectionEventHandlerProxy()
        {
            return connectionEventHandlerProxy;
        }
    }

}
